airbyte-ci/connectors/pipelines/pipelines/gradle.py

```diff
@@ -121,29 +122,69 @@ async def _run(self) -> StepResult:
                     ]
                 )
             )
-            # Set GRADLE_HOME and GRADLE_USER_HOME to the directory which will be rsync-ed with the gradle cache volume.
+            # Set GRADLE_HOME to the directory which will be rsync-ed with the gradle cache volume.
             .with_env_variable("GRADLE_HOME", "/root/.gradle")
+            # Same for GRADLE_USER_HOME.
             .with_env_variable("GRADLE_USER_HOME", "/root/.gradle")
             # Set RUN_IN_AIRBYTE_CI to tell gradle how to configure its build cache.
             # This is consumed by settings.gradle in the repo root.
             .with_env_variable("RUN_IN_AIRBYTE_CI", "1")
+            # Disable the Ryuk container because it needs privileged docker access which it can't have.
+            .with_env_variable("TESTCONTAINERS_RYUK_DISABLED", "true")
+            # Set the current working directory.
+            .with_workdir("/airbyte")
+        )
+
+        # Mount the whole git repo to update the persistent gradle cache and build the CDK.
+        with_whole_git_repo = (
+            gradle_container_base
+            # Mount the whole repo.
+            .with_mounted_directory("/airbyte", self.context.get_repo_dir("."))
+            # Mount the cache volume for the gradle cache which is persisted throughout all pipeline runs.
+            # We deliberately don't mount any cache volumes before mounting the git repo otherwise these will effectively be always empty.
```

Comment:
Why would it be empty as it's not sharing path with `/airbyte`?

Reply:
I honestly don't know why. Perhaps it's a dagger bug? In any case, if you mount the cache volume before the repo, that container layer gets aggressively cached by the dagger engine (as it should) and never gets rebuilt for subsequent airbyte-ci runs (which makes sense); but, somehow, this means that the cache volume is always empty on my laptop. Mounting it after the cache gets busted by a source file change makes it perform as expected.

Presumably, the same thing happens in our runners. It's weird and unexpected for sure.